models/contacts.js

```diff
@@ -1,19 +1,66 @@
-// const fs = require('fs/promises')
+const fs = require('fs/promises');
+const path = require('path');
+const { nanoid } = require("nanoid")
 
-const listContacts = async () => {}
+const contactsPath = path.join(__dirname, 'contacts.json');
 
-const getContactById = async (contactId) => {}
+const listContacts = async () => {
+  const data = await fs.readFile(contactsPath);
+  const contacts = JSON.parse(data.toString());
+  return contacts;
+};
 
-const removeContact = async (contactId) => {}
+const getContactById = async contactId => {
+  const contacts = await listContacts();
+  const contact = contacts.find(contact => {
+    return contact.id === contactId;
+  });
+  return contact || null;
+};
 
-const addContact = async (body) => {}
+const removeContact = async contactId => {
+  const contacts = await listContacts();
+  const index = contacts.findIndex(contact => contact.id === contactId);
+  const deletedContact = contacts[index];
+  if (index !== -1) {
+    contacts.splice(index, 1);
+    await fs.writeFile(contactsPath, JSON.stringify(contacts));
+  }
+  return deletedContact || null;
+};
 
-const updateContact = async (contactId, body) => {}
+const addContact = async body => {
+  const { name, email, phone } = body;
+  const newContact = {
+    id: nanoid(),
+    name: name,
+    email: email,
+    phone: phone,
+  };
+  const contacts = await listContacts();
+  contacts.push(newContact);
+  await fs.writeFile(contactsPath, JSON.stringify(contacts));
+  return newContact;
+};
+
+const updateContact = async (contactId, body) => {
+  const contacts = await listContacts();
+  const index = contacts.findIndex(contact => contact.id === contactId);
+  if (index !== -1) {
+    const contact = await getContactById(contactId);
+    contacts[index] = {
+      ...contact,
+      ...body,
+    };
+    await fs.writeFile(contactsPath, JSON.stringify(contacts));
```

Comment:
```suggestion
    await fs.writeFile(contactsPath, JSON.stringify(contacts, null, 2));
```
аналогічно в інших місцях, для форматування